superset-frontend/plugins/plugin-chart-echarts/src/MixedTimeseries/buildQuery.ts

```diff
@@ -67,11 +67,8 @@ export default function buildQuery(formData: QueryFormData) {
         fd,
         queryObject,
       )
-        ? timeComparePivotOperator(fd, queryObject)
-        : pivotOperator(fd, {
-            ...queryObject,
-            columns: fd.groupby,
-          });
+        ? timeComparePivotOperator(fd, baseQueryObject)
+        : pivotOperator(fd, baseQueryObject);
```

Comment:
The problem with using `baseQueryObject` is that it's the query builder's best guess as to what a single query query context would look like. In other words it will apply the same value to both queries, despite the fact that they should be separate. For this reason, we need to stick to whatever we have in `queryObject`, as that should already be rendered with the correct options. For this reason I'd actually propose the following change, which seems to work in all my quick test cases (only time compare on query A, only time compare on query B, and time compare on both queries):
```suggestion
        ? timeComparePivotOperator(fd, {
            ...queryObject,
            columns: fd.groupby,
          })
        : pivotOperator(fd, {
            ...queryObject,
            columns: fd.groupby,
          });
```

![image](https://user-images.githubusercontent.com/33317356/216262164-0229d155-824d-4b19-873c-83b44c284e13.png)

Reply:
Thanks for the suggestion, done